### PR TITLE
[Python] add python 3.7 and 3.8 to CI

### PR DESCRIPTION
**Pull request by rymurr**

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -45,10 +45,28 @@ matrix:
...
+    python:
+      - 3.7
+    before_script:
+      - cd python
+      - pip install -e .[dev]
+      - pip install -U tox-travis
+    script: tox
+  - language: python
+    python:
+      - 3.8
```
Comment (RussellSpitzer) [inline .travis.yml:65]: If all of these have the same before_scripts can't we just do 
```
python:
 -3.6
 -3.7
 -3.8
before_script:
script: 
```

Comment (rymurr) [reply .travis.yml:50]: I thought so as well. Travis seems to only run one python in that case.

Compare yesterdays travis run:
https://travis-ci.org/github/apache/iceberg/builds/720376892

To today's with the duplicate code removed:
https://travis-ci.org/github/apache/iceberg/builds/720565051

Comment (rdblue) [reply .travis.yml:66]: It looks fine with all 3 separate, but the 3.8 config lists both 3.6 and 3.7 as well. Should those be removed?

Comment (rymurr) [reply .travis.yml:64]: yes, fixed

Comment (RussellSpitzer) [reply .travis.yml:64]: Travis is a mystery sometimes :) I wouldn't waste time figuring it out since the current version works

**Comment by rdblue** [comment]
Thanks @rymurr! Merging this.

**Pull request merged**